pkg/core/mpt/helpers.go

```diff
@@ -14,7 +14,7 @@ func lcp(a, b []byte) []byte {
 	}
 
 	var i int
-	for i = 0; i < len(b); i++ {
+	for i = range len(b) {
```

Comment:
Can be `range b`.

Reply:
But tests fail. I vaguely remember I had to revert some and maybe it's one of those (but can't say why immediately).

Reply:
ok, got the reason but it was a little bit unexpected, usually, i am sad if such a simple thing requires some time to think, suggested refactoring

> Can be range b.

usually, i do not like such patterns cause i was born when python was already existing, but do not insist